classes/pickup/pickup.gd

```diff
@@ -85,7 +90,7 @@ func _pickup_id_setup() -> void:
 
 
 func _pickup_sound():
-	if sfx != null:
+	if sfx != null and has_node("SFXCollect"):
```

Comment:
Not a good idea.
Best case scenario, `_sfx_path` is left default, `sfx` already references `SFXCollect`, this check is redundant.
Worst case scenario, `_sfx_path` has been changed, no `SFXCollect` node even exists in the object, check can never succeed, entire object is permanently silent.

Reply:
The worst case scenario of a permanently silent object is preferable. Without the has_node check, sfx can and has been passed to new_from_existing, causing a crash (since it's already been freed from memory). 

The has_node check acts as a failsafe to prevent sfx being passed, that way the worst possible outcome is a silent pickup.

Reply:
That's fair.

In that case, try using `has_node(_sfx_path)` instead. That fixes both problems.

(And don't forget to comment your work, so future readers know what's going on here!)

Reply:
Wait--on second thought. 
Respawning is what causes the crash, right?
And with the other code change in place, this entire branch is unreachable when respawning is disabled, right?

So why do we even need to put a check here?

Reply:
It's definitely pointless now, but that `respawn_seconds` check is kind of fragile. If someone changed the code in a way that allowed `sfx` to be reparented and `respawn_seconds` to not be 0.0, it would start crashing again. 

Anyways, I added a `printerr()` to the code, that way any future crashes related to sfx will alert us to it's cause immediately.